src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -975,22 +1147,26 @@ static MemorySegment allocateNative(MemoryLayout layout) {
     }
 
     /**
-     * Creates a native memory segment with the given size (in bytes).
+     * Creates a native segment with the given size (in bytes).
+     * <p>
+     * The {@linkplain #address() address} of the returned memory segment is the starting address of
+     * the newly allocated off-heap region backing the segment. Moreover, the {@linkplain #address() address}
+     * of the returned segment is guaranteed to be at least 1-byte aligned.
      * <p>
      * The returned memory segment is associated with a new {@linkplain MemorySession#openImplicit implicit}
-     * memory session. As such, the native memory region associated with the returned segment is
+     * memory session. As such, the off-heap region which backs the returned segment is
      * freed <em>automatically</em>, some unspecified time after it is no longer referenced.
      * <p>
      * Native segments featuring deterministic deallocation can be obtained using the
-     * {@link MemorySession#allocate(long)}} method.
+     * {@link MemorySession#allocate(MemoryLayout)} method.
```

Comment:
Is this the right method to reference? Since this method takes a `long`, I'd expect the referenced method to be `allocate(long)` as well.